app/services/access_request_approval_service.rb

```diff
@@ -16,6 +16,13 @@ def call
 
     orgs_missing_on_target_user = @access_request.requester.organisations - target_user.organisations
     target_user.organisations << orgs_missing_on_target_user
+
+    # In addition, create the user permission relationship
+    missing_providers_on_target_user = orgs_missing_on_target_user.flat_map(&:providers)
+    missing_providers_on_target_user.each do |org_provider|
+      target_user.providers << org_provider
+    end
```

Comment:
Need a rebase on the Part 3 branch whenever you're ready, ie—we shouldn't be seeing this code in the diff.